src/GitInfo.php

```diff
@@ -53,24 +61,26 @@ public static function addCommand(string $name, $command)
      *
      * @return array
      */
-    public function getInfo(array $commands = [])
+    public function getInfo($commands = null)
     {
         $cwd = getcwd();
         chdir($this->getWorkingDirectory());
 
         $commandResult = [];
         // Only continue if we received any commands.
         if (!empty($commands)) {
-            foreach ($commands as $command) {
-                // Verify that the command is registered.
-                if (array_key_exists($command, self::$registeredCommands)) {
+            if (is_array($commands)) {
+                foreach ($commands as $command) {
                     // Execute the command and save the result to our array of commands.
-                    exec(self::$registeredCommands[$command], $result);
-                    $commandResult[$command] = $result;
-                    $result = [];
-                } else {
-                    throw new \Exception('Command: ' . $command . ' not registered.');
+                    $commandResult[$command] = $this->executeCommand($command);
                 }
+            } elseif (is_string($commands)) {
+                return $this->executeCommand($commands);
+            }
+        } else {
+            // Execute all the commands registered.
```

Comment:
Please add a test to test for when `gitInfo()` is called without a `$commands` parameter.